src/urllib3/contrib/socks.py

```diff
@@ -66,23 +67,37 @@
 try:
     import ssl
 except ImportError:
-    ssl = None
+    ssl = None  # type: ignore
+
+try:
+    from typing import TypedDict
+except ImportError:
+    from typing_extensions import TypedDict  # Python 3.7
```

Comment:
We don't depend on `typing_extensions` so can't use it here. In this case we can set `_TYPE_SOCKS_OPTIONS` to `Dict[str, Any]`? Is mypy on Python 3.6 okay with that?

Reply:
This PR actually highlights that we maybe shouldn't have typing-extensions in dev-requirements.txt if it's not being used because this failure wouldn't be caught in CI until release integration tests.

Reply:
I can remove it in a seperate PR

Reply:
Here is the PR https://github.com/urllib3/urllib3/pull/2275